### PR TITLE
Configure ref- and staging- managers on entry cataloger

### PR DESCRIPTION
**Pull request by arielshaqed**
No committed manager yet, but all(?) the building blocks are right there with a TODO.

Clean up all cataloger construction, which can now fail.

### CODE REVIEW

**Comment by arielshaqed** [comment]
- @nopcoder + @guy-har due to availability.  Still welcome to review, of course!

File: catalog/rocks/cataloger.go [hunk 1]
```diff
@@ -25,13 +27,14 @@ const (
...
+func NewCataloger(db db.Database, cfg *config.Config) (catalog.Cataloger, error) {
+	entryCatalog, err := NewEntryCatalog(cfg, db)
 	return &cataloger{
-		EntryCatalog: NewEntryCatalog(),
+		EntryCatalog: entryCatalog,
 		log:          logging.Default(),
 		dummyDedupCh: make(chan *catalog.DedupReport),
 		hooks:        catalog.CatalogerHooks{},
-	}
+	}, err
```
Comment (guy-har) [inline catalog/rocks/cataloger.go:37]: return nil in case of error

**Review by guy-har — APPROVED** [review]
looks great :)

File: catalog/rocks/cataloger.go (hunk 1 above)
Comment (arielshaqed) [reply catalog/rocks/cataloger.go]: OK.

**Review by arielshaqed — COMMENTED** [review]
Thanks!  Pushing once tests pass.

**Pull request merged**